### PR TITLE
refactor(`RunMigrations`): func that copies, migrates, and moves the ipfs

### PR DESCRIPTION
**Pull request by ramfox**
`func RunMigrations(oldpath, newpath string) error`
- takes an old path
- copies the repo to a temp dir
- migrates the copied repo
- moves the migrated repo to the new location
- cleans up any temp dir
If there is any error, it removes the new repo 

**doesn't remove the original repo**

Created a small util program that moves and migrates repo and it works!!

### CODE REVIEW

File: cafs/ipfs/config.go [hunk 1]
```diff
@@ -118,16 +117,16 @@ func (cfg *StoreCfg) InitRepo(ctx context.Context) error {
 // MoveIPFSRepoOntoPath moves the ipfs repo from wherever it is,
 // indicated by the store config, to live on the given path
 // this changes the path in the given config struct
-func MoveIPFSRepoOntoPath(o *StoreCfg, path string) error {
-	if path == "" {
+func MoveIPFSRepoOntoPath(repoPath, newPath string) error {
```
Comment (dustmop) [inline cafs/ipfs/config.go]: This is equivalent to just `os.Rename`. If either argument to that function is an empty string you'll get the error "no such file or directory". How about calling that directly from `RunMigrations`.

**Review by dustmop — APPROVED** [review]
LGTM, just have one suggested change

File: cafs/ipfs/config.go (hunk 1 above)
Comment (ramfox) [reply cafs/ipfs/config.go]: Yea that sounds good. We've minimized/changed the responsibilities of `MoveIPFSRepoOntoPath` so much that it has slimmed down so much as to be unnecessary.

**Review by dustmop — APPROVED** [review]
LGTM

File: cafs/ipfs/migrate.go [hunk 2]
```diff
@@ -2,14 +2,65 @@ package ipfs_filestore
...
+	// create temp directory into which we will copy the
+	// ipfs directory
+	tmpDir, err := ioutil.TempDir(os.TempDir(), "ipfs_temp")
+	if err != nil {
+		return fmt.Errorf("error creating temp directory: %s", err)
+	}
+
+	rollback := func() {
+		if ipfsRepoPath != newRepoPath {
+			os.RemoveAll(newRepoPath)
```
Comment (Arqu) [inline cafs/ipfs/migrate.go:32]: RemoveAll is usually pretty scary but don't have any input on how to make it better in a sane way.

**Review by Arqu — APPROVED** [review]
👍

**Pull request merged**